pkg/tenant/tenant.go

```diff
@@ -177,7 +178,22 @@ func (s *Server) Shutdown() (err error) {
 }
 
 // Sets up the server's middleware and routes
-func (s *Server) setupRoutes() error {
+func (s *Server) setupRoutes() (err error) {
+	// Set the authentication overrides from the configuration
+	opts := mw.AuthOptions{
+		Audience: s.conf.Auth.Audience,
+		Issuer:   s.conf.Auth.Issuer,
+		KeysURL:  s.conf.Auth.KeysURL,
+	}
+
+	// In maintenance mode authentication is disabled
+	var authenticator gin.HandlerFunc
+	if !s.conf.Maintenance {
```

Comment:
I think the problem was that the maintenance mode test wasn't starting an authtest server so the code here was returning an error. However if we're worried about the nil middleware I can just update the test instead of having the check here.

Reply:
Perfect, I'd prefer that we update the test - thank you!